tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/core/portal/PortalSourcePathComputerDelegate.java

```diff
@@ -54,6 +57,41 @@
 
 		IServer server = ServerUtil.getServer(configuration);
 
+		PortalServerBehavior behavior = (PortalServerBehavior)server.loadAdapter(PortalServerBehavior.class, monitor);
```

Comment:
this is for watch projects, but does this also support everything else, normally deployed projects?  (not with watch but with just 'configured' projects)?

Reply:
This only support user manually watch project(This project can be a child project of liferay gradle workspace or a standard alone gradle project. if a watched gradle project is also placed in liferay gradle workspace, it will be added twice. But the _addSourceContainers will be ignore it, so I didn't do any check here.)  For normal deploy project, I feel it can support, but I manually trigger sourcePathComputer add watch project to source lookup containers.